assignment3/assignment3.cpp

```diff
@@ -6,191 +7,158 @@
 #include <set>
 
 class Dictionary{
-	std::vector<std::string> dictionary;
+    std::vector<std::string> dictionary;
 public:
-	Dictionary(const std::string &filemame){
-		std::ifstream file (filemame);
-  		if (file.is_open()) {
- 			std::string line;
-    		while (std::getline (file,line)) {
-      			dictionary.push_back(line);
-    		}
-    		file.close();
-  		}
-	}
-	bool isPrefix(const std::string &word) const {
-		bool is_prefix = false;
-		for(int i = 0; i<dictionary.size() && !is_prefix; i++){
-			is_prefix = true;
-			for(int j = 0; j<word.size() && is_prefix; j++){
-				if(j<dictionary[i].size()){
-					is_prefix = dictionary[i][j] == word[j];
-				} else {
-					is_prefix=false;
-				} 
-			}
-		}
-		return is_prefix;
-
-	}
-	bool isWord(const std::string &word) const{
-		bool is_word = false;
-		for(int i = 0; i<dictionary.size() && !is_word; i++){
-			is_word = dictionary[i].compare(word) == 0;
-		}
-		return is_word;
-	}
+    Dictionary(const std::string &filemame){
+        std::ifstream file (filemame);
+        if (file.is_open()) {
+            std::string line;
+            while (std::getline (file,line)) {
+                dictionary.push_back(line);
+            }
+            file.close();
+        } else {
+            dictionary.clear();
```

Comment:
I think it will be good to either inform the user that the file couldn't be open (including the filename in the message), or throw an exception (but this is not used much).
This clear will have no effect, right? We are in the constructor and haven't added anything to the dictionary yet.